c2/cuda/groth16_srs.cuh

```diff
@@ -201,48 +212,73 @@ private:
             cursor += read_g1_point(&vk.delta_g1, srs_ptr + cursor);
             cursor += read_g2_point(&vk.delta_g2, srs_ptr + cursor);
 
-            assert(file_size > cursor + sizeof(uint32_t));
+            if (file_size <= cursor + sizeof(uint32_t)) {
+                munmap(const_cast<byte*>(srs_ptr), file_size);
+                throw sppark_error{-EOF, "SRS file size/layout mismatch"};
```

Comment:
EOF is not an error code, but a "marker" used by stdio. In the context it would be appropriate to choose among codes returned by `lseek`. Because indexing an mmap-ed file is similar to lseek-ing and read-ing.